clsp/fusion_singleton.clsp

```diff
@@ -39,26 +39,22 @@
       ; who am I?
       (list ASSERT_MY_COIN_ID singleton_coin_id)
       (list ASSERT_MY_PUZZLEHASH (calculate_full_puzzle_hash SINGLETON_MOD_HASH LAUNCHER_ID LAUNCHER_PUZZLE_HASH singleton_inner_puzzlehash))
-      (list ASSERT_MY_AMOUNT 1)
+      (list ASSERT_MY_AMOUNT 1) ; This feels unnecessary. And if you ever create such a singleton with amount > 1 it is locked forever.
 
       ; spend myself forward
       (list CREATE_COIN singleton_inner_puzzlehash 1 (list LAUNCHER_ID)) ;; opt to keep details beyond launcher ID encoded in eve coin spend
     )
   )
 
   ; maintain announcement loop with p2 for unlocking to puzzlehash
-  (defun announce_p2_releases (singleton_coin_id nft_ids_to_release release_puzzlehash conditions_so_far)
-      (if (l (r nft_ids_to_release))
-        (announce_p2_releases singleton_coin_id (r nft_ids_to_release) release_puzzlehash
-          (c 
-            (list CREATE_PUZZLE_ANNOUNCEMENT (sha256 singleton_coin_id (f (f nft_ids_to_release)) release_puzzlehash))
-            conditions_so_far
-          )
-        )
-        (c 
+  ; You can simplify those for loops a lot by turning them upside down:
```

Comment:
I feel dense, but I'm not quite following here. Could you describe the mechanism a bit more?